examples/font-info.rs

```diff
@@ -84,6 +86,94 @@ fn main() {
         }
     }
 
+    if let Some(stat) = face.tables().stat {
```

Comment:
Let's put it into a separate functions, like with `print_opentype_layout`.

Also, it's pretty verbose. Maybe there is a way to make the code shorter. Maybe we should expose a more high-level API as well, to avoid direct subtables accesses? Would it be doable?

Reply:
> Maybe we should expose a more high-level API as well, to avoid direct subtables accesses? 

Well anything's possible.  What did you have in mind?

Ignoring the format 4 tables, for a non-variable font I imagine querying by axis tag and getting a single value back makes sense.  For instance Robto Condensed lists a "normal" width in the `OS/2` table, but lists the proper width in the `STAT` table.  So a width check on a non-variable font might look first in `STAT` to see if there's an entry for `wdth` and grab whatever from the `OS/2` table as a fallback.

For a variable font, the query would be by tag and value (based on what's in `fvar`).  E.g. what's the name for `wght` == 500?

Format 4 tables open up the possibility of querying via multiple axes (e.g. what's the stretch at `wght` == 200 and `ital` == 1.0), but that's something I'd prefer to punt on for now.

```rust

enum StyleValue {
  Single(Fixed),
  Range { min: Fixed, max: Fixed, nominal: Fixed },
  Link { value: Fixed, target: Fixed },
}

struct StyleAttribute {
  name: Tag,
  flags: Flags,
  value: StyleValue,
}

fn style_for_axis(tag: Tag) -> StyleAttribute;
fn style_for_axis_and_value(tag: Tag, value: Fixed) -> StyleAttribute;

// Only useful with format 4 tables
fn style_for_values(values: &[(Tag, Fixed)]) -> StyleAttribute;
```

Reply:
I never looked into how the STAT table looks, so I cannot comment for now. I would have to investigate it myself.
I was just curious if the code in this example can be simplified. Right now it's too big for a simple example.

Reply:
Welp I'm all for exposing a higher level query API.  Agreed that what's in the example is too low level.

Reply:
Let's move the STAT code to a new function for no, as I have suggested earlier. And leave it as is otherwise for now.

Reply:
Ah, by the time I saw this I'd already hacked up a simple query function which I think cleans things up a little bit and creates a foundation to update `is_italic` and `is_bold`.